AsyncDisplayKit/ASCollectionView.mm

```diff
@@ -411,7 +409,9 @@ - (void)setAsyncDelegate:(id<ASCollectionViewDelegate>)asyncDelegate
 
   super.delegate = (id<UICollectionViewDelegate>)_proxyDelegate;
   
-  [_layoutInspector didChangeCollectionViewDelegate:asyncDelegate];
+  if ([_layoutInspector respondsToSelector:@selector(didChangeCollectionViewDelegate:)]) {
```

Comment:
It's a pretty cheap call as I guess setting the delegate will not happen that often, but I will move it to a cached flag